src/Microsoft.ML.Data/Transforms/ConversionsExtensionsCatalog.cs

```diff
@@ -141,5 +141,19 @@ public static KeyToValueMappingEstimator MapKeyToValue(this TransformsCatalog.Co
             IEnumerable<TOutputType> values,
             params (string source, string name)[] columns)
             => new ValueMappingEstimator<TInputType, TOutputType>(CatalogUtils.GetEnvironment(catalog), keys, values, columns);
+
+        /// <summary>
+        /// Maps specified keys to specified values
+        /// </summary>
+        /// <param name="catalog">The categorical transform's catalog</param>
+        /// <param name="lookupMap">An instance of <see cref="IDataView"/> that contains the key and value columns.</param>
+        /// <param name="keyColumn">Name of the key column in <paramref name="lookupMap"/>.</param>
+        /// <param name="valueColumn">Name of the value column in <paramref name="lookupMap"/>.</param>
+        /// <param name="columns">The columns to apply this transform on.</param>
+        /// <returns></returns>
+        public static ValueMappingEstimator ValueMap(
+            this TransformsCatalog.ConversionTransforms catalog,
+            IDataView lookupMap, string keyColumn, string valueColumn, params (string input, string output)[] columns)
```

Comment:
@TomFinley @sfilipi - is this consistent with the order we've decided on with #2064?

Reply:
Any thoughts guys? I saw the method above has same pattern so I followed that.

---
In reply to: [252014795](https://github.com/dotnet/machinelearning/pull/2302#discussion_r252014795) [](ancestors = 252014795)